src/config/localfile-config.c

```diff
@@ -46,8 +43,56 @@ int Read_Localfile(XML_NODE node, void *d1, __attribute__((unused)) void *d2)
     logreader_config *log_config;
     size_t labels_z=0;
     label_flags_t flags;
+    wm_logcollector_t *logcollector = NULL;
+    wmodule **wmodules = (wmodule**)d1;
+    wmodule *cur_wmodule;
+
+    // Allocate memory
+    if ((cur_wmodule = *wmodules)) {
+        wmodule *cur_wmodule_exists = *wmodules;
+        int found = 0;
+
+        while (cur_wmodule_exists) {
+            if(cur_wmodule_exists->tag) {
+                if(strcmp(cur_wmodule_exists->tag,LOGCOLLECTOR_WM_NAME) == 0) {
+                    cur_wmodule = cur_wmodule_exists;
+                    found = 1;
+                    break;
+                }
+            }
+            cur_wmodule_exists = cur_wmodule_exists->next;
+        }
 
-    log_config = (logreader_config *)d1;
+        if(!found) {
+            while (cur_wmodule->next) {
+                cur_wmodule = cur_wmodule->next;
+            }
+
+            os_calloc(1, sizeof(wmodule), cur_wmodule->next);
+            cur_wmodule = cur_wmodule->next;
+            cur_wmodule->context = &WM_LOGCOLLECTOR_CONTEXT;
+            cur_wmodule->tag = strdup(LOGCOLLECTOR_WM_NAME);
+        }
+    } else {
+        *wmodules = cur_wmodule = calloc(1, sizeof(wmodule));
+        cur_wmodule->context = &WM_LOGCOLLECTOR_CONTEXT;
+        cur_wmodule->tag = strdup(LOGCOLLECTOR_WM_NAME);
+    }
+
+    if (NULL == cur_wmodule->data) {
+        os_calloc(1, sizeof(wm_logcollector_t), logcollector);
+        /* Reading the internal options */
+        int ret = OS_SUCCESS;
+        if (ret = wm_logcollector_read(logcollector), OS_SUCCESS != ret) {
```

Comment:
maybe you can do this in one single line to avoid an extra assignment. 
I mean:
```
const int ret = wm_logcollector_read(logcollector);
if (OS_SUCCESS != ret) {
   ...
}
```